pkg/workflows/dontime/request.go

```diff
@@ -0,0 +1,65 @@
+package dontime
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type Request struct {
+	ExpiresAt time.Time
+
+	// CallbackCh is a channel to send a response back to the requester
+	// after the request has been processed or timed out.
+	CallbackCh chan Response
+
+	WorkflowExecutionID string
+	SeqNum              int
+}
+
+func (r *Request) ID() string {
+	return r.WorkflowExecutionID
+}
+
+func (r *Request) ExpiryTime() time.Time {
+	return r.ExpiresAt
+}
+
+func (r *Request) SendResponse(_ context.Context, resp Response) {
+	select {
+	case r.CallbackCh <- resp:
+		close(r.CallbackCh)
+	default: // Don't block trying to send
```

Comment:
nit: this should never happen as we need to guarantee the channel is large enough - maybe add an Error log?